### ISSUE TITLE
Provide a set of endpoints for user management

### ISSUE THREAD

**Issue by syntrydy** [JanssenProject/jans#418]
### User Story Title.

Implement endpoints for user management

### User Story Meta Information.

This should be base on the existing SCIM project

#### Priority.

 MEDIUM

### Description.

As an API consumer with the user management permission set, I should be able to: 
 
- List all users
- Search for users base on a search pattern
- Add new user
- Update existing user
- Remove a user 
-  Activate/Deactivate a user
- Manage user password:
            1. Password policy:  define the password complexity(eg. min length, 
            2. Password Reset: reset my password
            3. Username Recovery

**Comment by yurem** [comment]
I think we should not expose user API in this application. Why we can't use SCIM and configure it to conform our requirements?
We should keep minimum count of service which can expose user data.
@nynymike

**Comment by nynymike** [comment]
I agree with Yura. What is the advantage to exposing endpoints that just duplicate the functionality of SCIM?

**Comment by duttarnab** [comment]
@yurem @nynymike , SCIM apis does not provide all attributes from `jansPerson` table. As @jgomer2001 has mentioned

![image](https://user-images.githubusercontent.com/32794267/160332551-57d72307-cf47-4eb3-8fd0-8d2bf4763a1e.png)
![image](https://user-images.githubusercontent.com/32794267/160332626-fd527de5-53c7-4dea-a29a-d905fb9409ae.png)


Moreover, if a user manually updates the schema to add a new attribute to `jansPerson` table then this attribute will also not be available using SCIM apis as it does not provide all attributes.

That's why we are trying to create new User Management APIs in config-api.

**Comment by yurem** [comment]
After manual schema update did you register this attribute in `ou=attributes`? In Jans CE setup there are scim test data. You can use `-t` option to prepare test server. In `scim-test-data.ldif` setup registers 3 custom attributes which uses in tests.

**Comment by duttarnab** [comment]
We need to remove `password` from the response of the below endpoints because if we have the user password in the get endpoint response then it can be easily traced from the browser console.

- GET /jans-config-api/api/v1/users (get all users)
- GET /jans-config-api/api/v1/users/{inum} (get a user by inum)

**Comment by pujavs** [comment]
**Implemented:** 
- PATCH endpoint -> /jans-config-api/api/v1/user/{inum}
- Exclusion of `userPassword` in fetch request
  - GET /jans-config-api/api/v1/users (get all users)
  - GET /jans-config-api/api/v1/users/{inum} (get a user by inum)

**Test results:**
[src.test.resources.feature.user.user.pdf](https://github.com/JanssenProject/jans/files/8454281/src.test.resources.feature.user.user.pdf)

**Comment by pujavs** [comment]
Post discussion with @nynymike and @jgomer2001, since adding all 'jansPerson` attributes to SCIM api is not feasible and correct it is decided to **remove the `SCIM - User Management` endpoint and convert the new config `User Management` endpoint into plugin.**

![image](https://user-images.githubusercontent.com/43700552/163226574-febbabc9-63f2-4291-ab45-749993f0f29c.png)

Refer older comment from this issue as well:
https://github.com/JanssenProject/jans/issues/418#issuecomment-1080216424

**Comment by pujavs** [comment]
@duttarnab changes done for following 
1) Removing scim user management endpoint
2) Moving user management endpoint in user-mgt-plugin.
Note change in endpoint url, new url is `/jans-config-api/mgt/configuser`

**Comment by pujavs** [comment]
Issue: For user management endpoints `mail,displayName,jansStatus,userPassword,givenName` have been made mandatory fields and the endpoint **does not return** `userPassword`.
This will work fine for POST - create user request.
However since 'userPassword` is not returned the PUT methods throws error if `userPassword` is not provided. 
Example: `jans-cli` uses swagger spec in which the mandatory fields have to be specified at the object level.
So the PUT request throws error as it does not contain `userPassword` 
**To resolve this it is decided to make `userPassword` mandatory only for POST method**

**Comment by pujavs** [comment]
This requirement and related enhancements for root level attributes and mandatory attribute check was implemented on 08-Jun-2022 hence closing the same.
